src/kvstore/DiskManager.h

```diff
@@ -90,37 +93,41 @@ class DiskManager {
    * @return true Data path remains enough space
    * @return false Data path does not remain enough space
    */
-  bool hasEnoughSpace(GraphSpaceID spaceId, PartitionID partId);
+  bool hasEnoughSpace(GraphSpaceID spaceId, PartitionID partId) const;
 
   /**
    * @brief Get all partitions grouped by data path and spaceId
    *
    * @param diskParts Get all space data path and all partition in the path
    */
-  void getDiskParts(SpaceDiskPartsMap& diskParts);
+  void getDiskParts(SpaceDiskPartsMap& diskParts) const;
 
  private:
   /**
    * @brief Refresh free bytes of data path periodically
    */
   void refresh();
 
+  struct Paths {
+    // canonical path of data_path flag
+    std::vector<boost::filesystem::path> dataPaths_;
+    // given a space and data path, return all parts in the path
+    std::unordered_map<GraphSpaceID, PartDiskMap> partPath_;
+    // the index in dataPaths_ for a given space + part
+    std::unordered_map<GraphSpaceID, std::unordered_map<PartitionID, size_t>> partIndex_;
+  };
+
  private:
   std::shared_ptr<thread::GenericWorker> bgThread_;
 
-  // canonical path of data_path flag
-  std::vector<boost::filesystem::path> dataPaths_;
+  std::atomic<Paths*> paths_;
   // free space available to a non-privileged process, in bytes
   std::vector<std::atomic_uint64_t> freeBytes_;
 
-  // given a space and data path, return all parts in the path
-  std::unordered_map<GraphSpaceID, PartDiskMap> partPath_;
-
-  // the index in dataPaths_ for a given space + part
-  std::unordered_map<GraphSpaceID, std::unordered_map<PartitionID, size_t>> partIndex_;
-
   // lock used to protect partPath_ and partIndex_
   std::mutex lock_;
+
+  mutable folly::rcu_domain<> rcuDomain_;
```

Comment:
Maybe you don't need to declare a `rcu_domain` explicitly, the default one is enough.

Reply:
> Maybe you don't need to declare a `rcu_domain` explicitly, the default one is enough.

metaclient also use default, they share the same one?